### PR TITLE
rework navigation

### PR DESCRIPTION
**Pull request by dominictarr**
This is quite a big change, but I refactored the nav out into it's own module
http://github.com/dominictarr/hypernav
this reverses the pages, so each page just provides it's contents, not it's navigation, the app controls navigation. router, is still used. This also incorporates the history api.

I'm very happy with how snappy it is!

### CODE REVIEW

**Comment by dominictarr** [comment]
oh, I also took the liberty of moving all static strings I could find into translations

File: app/page/userShow.js [hunk 1]
```diff
@@ -12,25 +13,29 @@ exports.needs = nest({
...
 
     const { feed } = location
     const Link = api.app.html.link
     const myId = api.keys.sync.id()
+    var name = api.about.html.name(feed)
 
-    return h('Page -userShow', [
-      h('h1', api.about.obs.name(feed)),
-      api.app.html.nav(),
+    return h('Page -userShow', {title: api.about.obs.name(feed)}, [
```
Comment (mixmix) [inline app/page/userShow.js]: want to talk about translation strings. I've got some experience I'd like to inform how we do this

File: app/html/app.js [hunk 2]
```diff
@@ -3,46 +3,54 @@ const values = require('lodash/values')
...
+      api.app.async.catchLinkClick(document.body, (link, { isExternal }) => {
+        if (isExternal) return openExternal(link)
+        nav.push(link)
+      })
+      nav.push({page: 'home'})
+      return nav
+    },
+    'history.sync.push': (location) => nav.push(location),
+    'history.sync.back': () => nav.back(),
+    'history.obs.location': () => nav.history,
```
Comment (mixmix) [inline app/html/app.js]: why not just re-write `patch-history`?
I was keen to see what you single-file solution looked like

**Comment by mixmix** [comment]
rm `patch-history` as a dep

File: app/html/app.js (hunk 2 above)
Comment (dominictarr) [inline app/html/app.js]: actually, i did that https://github.com/mixmix/patch-history/pull/1

**Comment by mixmix** [comment]
great work!
super snappy. I've styled the nav bar to be pinned to the top of the page.
MERGED

**Comment by dominictarr** [comment]
great, two improvement ideas: 

* store and restore scroll positions
* pass already loaded data to when creating new screens, so that there is something ready to display immediately

**Comment by mixmix** [comment]
I'll put those suggestions in a card in backlog

**Pull request merged**